### DISCUSSION TITLE
Prevent the `x` in `\frac{x}{}` to become `xcolor` when typing `Tab`.

### DISCUSSION THREAD

**Discussion by jademath** [James-Yu/LaTeX-Workshop#4216] [Q&A]
In the default settings, when I input `\fr`, press `Tab`, type `x`, and then press `Tab` again, the system auto-completes `x` to `xcolor`. This is not the desired outcome because my intention is to use `Tab` to move to the next set of braces. I found a workaround by enabling 'tab completion' and simultaneously disabling 'quick suggestions'. This solution is almost perfect, except for one issue: typing `beq` followed by `Tab` no longer works. Instead, I have to type `BEQ` (in uppercase) followed by `Tab` to create the desired environment.

Ideally, I'd like to have 'quick suggestions' enabled only for snippet keywords, such as `beq`, `mbf`, and so on. However, I am unsure how to achieve this. While there is an option to enable 'tab completion' for snippets only, a similar option for 'quick suggestions' seems to be lacking.

**Comment by jademath** [comment]
A quick fix I can think of is to define both upper and lower-case snippets. Is this already implemented?

**Comment by jlelong** [comment]
You document most likely contains the word `xcolor`. Typing `x` inside `\frac{}{}` triggers quick suggestions and hitting `TAB` accepts the currently selected quick selection. The are two ways to tweak this behaviour

- Set `editor.suggest.snippetsPreventQuickSuggestions` to `true` to prevent quick suggestions to show up when a snippet is already active. 
- You can also play with `editor.suggest.selectionMode` to decide when you want a snippet to be automatically selected when a suggestion list shows up.
